AWSClientRuntime/Sources/Regions/DefaultRegionResolver.swift

```diff
@@ -27,14 +27,10 @@ public struct DefaultRegionResolver: RegionResolver {
     
     public func resolveRegion() async -> String? {
         for provider in providers {
-            do {
-                logger.debug("Attempting to resolve region with: \(String(describing: type(of: provider)))")
-                if let region = try await provider.resolveRegion() {
-                    logger.debug("Resolved region with: \(String(describing: type(of: provider)))")
-                    return region
-                }
-            } catch {
-                return nil
+            logger.debug("Attempting to resolve region with: \(String(describing: type(of: provider)))")
+            if let region = try? await provider.resolveRegion() {
```

Comment:
If this call throws, do we need to do something more than log failure?  With `try?` we don't even see the cause of the failure.

Reply:
We need to escape the error here somehow. The best we can do is log the error, which i can add in if that's desired. Otherwise, if we catch and return nil then we may not resolve a region if an earlier resolver fails. We should not fail, and instead move on to the next resolver in the list. (im looking for a SEP to verify this is indeed the intended behavior)

Reply:
Verified the intended behavior in slack.